src/software/new_geom/circle_test.cpp

```diff
@@ -0,0 +1,73 @@
+#include "software/new_geom/circle.h"
+
+#include <gtest/gtest.h>
+
+TEST(CreateCircleTests, circle_default_constructor)
+{
+    Circle c = Circle();
+    EXPECT_EQ(0, c.getRadius());
+    EXPECT_EQ(Point(), c.getOrigin());
+}
+
+TEST(CreateCircleTests, circle_custom_constructor)
+{
+    Circle c = Circle(Point(1, 2), 5);
+    EXPECT_EQ(5, c.getRadius());
+    EXPECT_EQ(Point(1, 2), c.getOrigin());
+
+    EXPECT_THROW(Circle(Point(3, 5), -4), std::invalid_argument);
+}
+
+TEST(CircleLogicTests, circle_getter_setter_tests)
+{
+    Circle c = Circle();
+    c.setOrigin(Point(2, -3));
+    c.setRadius(10);
+    EXPECT_EQ(Point(2, -3), c.getOrigin());
+    EXPECT_EQ(10, c.getRadius());
+    c.setOrigin(Point(-6, 7));
+    EXPECT_EQ(Point(-6, 7), c.getOrigin());
+
+    EXPECT_THROW(c.setRadius(-1), std::invalid_argument);
+}
```

Comment:
Personally I'd split these into `circle_set_origin` `circle_set_valid_radius` `circle_set_invalid_radius`, etc., so each test is testing something very specific. I don't feel super strongly about it here though, so totally optional.

Reply:
Similar comment goes for the below tests. Generally we want to be testing one specific thing per-test, not mutating things and running multiple asserts that check different things. It really does help when someone has to come back and read the tests (especially when the classes are more complicated then `Circle`).

Reply:
Fixed in latest